Sources/NIO/SelectorGeneric.swift

```diff
@@ -121,18 +121,15 @@ protocol _SelectorBackendProtocol {
 /* this is deliberately not thread-safe, only the wakeup() function may be called unprotectedly */
 internal class Selector<R: Registration>  {
     var lifecycleState: SelectorLifecycleState
-
-    // temporary workaround to stop us delivering outdated events; read in `whenReady`, set in `deregister`
-    var deregistrationsHappened: Bool = false
     var registrations = [Int: R]()
+    var sequenceIdentifier : RegistrationSequenceIdentifier = 1
 
-    let externalSelectorFDLock = Lock()
-
+    let myThread: NIOThread
     // The rules for `self.selectorFD`, `self.eventFD`, and `self.timerFD`:
     // reads: `self.externalSelectorFDLock` OR access from the EventLoop thread
     // writes: `self.externalSelectorFDLock` AND access from the EventLoop thread
+    let externalSelectorFDLock = Lock()
```

Comment:
These things got moved around a bit: can they be put back? Helps keep the `git blame` a bit more valuable.